scrc/dataset_creation/pretraining_dataset_creator.py

```diff
@@ -37,9 +37,9 @@ def prepare_dataset(self, save_reports) -> datasets.Dataset:
             "judgment": False, "citation": False, "lower_court": False
         }
 
-        for court_string in court_strings:
+        for string in court_strings:
             # we don't use the cache since it is overwritten after each court
-            df = self.get_df(engine, data_to_load, court_string=court_string, use_cache=False)
+            df = self.get_df(engine, data_to_load, court_string=string, use_cache=False)
 
             print_memory_usage([df, hf_dataset])
 
```

Comment:
I guess here it the loop needs to be removed, since this is done in the dataset_creator now. So here just the entire court_string list should be passed and then the concatenate option set to True

Reply:
Alright, should I still keep the garbage collector and the `print memory usage()`-statement?

Reply:
That's a good question. I don't know if it really makes a difference. So I guess you can keep it, but maybe add a TODO: saying that it should be investigated if it is really necessary.

Reply:
That's a good question. I don't know if it really makes a difference. So I guess you can keep it, but maybe add a TODO: saying that it should be investigated if it is really necessary.

Reply:
👍 I moved them to the place where the concatenating happens. I assume this is the potential critical process for the memory.

Reply:
Yes, exactly. Thanks